skimage/io/tests/test_io.py

```diff
@@ -53,8 +44,10 @@ def test_imread_file_url():
     assert image.shape == (512, 512)
 
 
-def test_imread_http_url(urllib_urlopen):
-    query_string = '?' + 's' * 266
-    image_url = 'https://image.com/camera.jpg{}'.format(query_string)
-    image = io.imread(image_url)
+def test_imread_http_url(httpserver):
+    # httpserver is a fixture provided by pytest-localserver
+    # https://bitbucket.org/pytest-dev/pytest-localserver/
+    httpserver.serve_content(one_by_one_jpeg)
+    # image = io.imread(httpserver.url + '?' + 's' * 266)
+    image = io.imread(httpserver.url + '/test.jpg' + '?' + 's' * 266)
```

Comment:
(a) where does the "test.jpg" part of the URL come from?
(b) why is the comment above this line still sticking around?

Reply:
ooooooh! 👏 

Sometimes pytest really is too magical. Thanks for the clarification, it's super useful!

Reply:
right....